analyzers/src/SonarAnalyzer.CSharp/Rules/VariableUnused.cs

```diff
@@ -42,17 +43,29 @@ public sealed class VariableUnused : VariableUnusedBase
             {
                 var collector = new UnusedLocalsCollector();
 
-                cbc.RegisterSyntaxNodeAction(collector.CollectDeclarations, SyntaxKind.LocalDeclarationStatement);
+                cbc.RegisterSyntaxNodeAction(collector.CollectDeclarations, SyntaxKind.LocalDeclarationStatement, SyntaxKind.SimpleAssignmentExpression);
                 cbc.RegisterSyntaxNodeAction(collector.CollectUsages, SyntaxKind.IdentifierName);
                 cbc.RegisterCodeBlockEndAction(collector.GetReportUnusedVariablesAction(Rule));
             });
 
-        private class UnusedLocalsCollector : UnusedLocalsCollectorBase<LocalDeclarationStatementSyntax>
+        private sealed class UnusedLocalsCollector : UnusedLocalsCollectorBase<SyntaxNode>
         {
-            protected override IEnumerable<SyntaxNode> GetDeclaredVariables(LocalDeclarationStatementSyntax localDeclaration) =>
-                localDeclaration.UsingKeyword().IsKind(SyntaxKind.UsingKeyword)
-                ? Enumerable.Empty<SyntaxNode>()
-                : localDeclaration.Declaration.Variables;
+            protected override IEnumerable<SyntaxNode> GetDeclaredVariables(SyntaxNode variableDeclaration)
+            {
+                if (variableDeclaration as LocalDeclarationStatementSyntax is { } localDeclaration)
+                {
+                    return localDeclaration.UsingKeyword().IsKind(SyntaxKind.UsingKeyword)
+                    ? Enumerable.Empty<SyntaxNode>()
+                    : localDeclaration.Declaration.Variables;
+                }
+                else
+                {
+                    var assignmentExpression = (AssignmentExpressionSyntax)variableDeclaration;
+                    return TupleExpressionSyntaxWrapper.IsInstance(assignmentExpression.Left)
+                           ? assignmentExpression.AssignmentTargets()
+                           : Enumerable.Empty<SyntaxNode>();
+                }
+            }
```

Comment:
In the assignment expression, we are only interested in declared variables here, so you need to filter further. I would also prefer a switch expression.  e.g like so:
```suggestion
            protected override IEnumerable<SyntaxNode> GetDeclaredVariables(SyntaxNode variableDeclaration) =>
                variableDeclaration switch
                {
                    LocalDeclarationStatementSyntax localDeclaration when !localDeclaration.UsingKeyword().IsKind(SyntaxKind.UsingKeyword) =>
                        localDeclaration.Declaration.Variables,
                    AssignmentExpressionSyntax assignmentExpression =>
                        assignmentExpression.AssignmentTargets().Where(x => DeclarationExpressionSyntaxWrapper.IsInstance(x) || SingleVariableDesignationSyntaxWrapper.IsInstance(x)),
                    _ => Enumerable.Empty<SyntaxNode>(),
                };
```